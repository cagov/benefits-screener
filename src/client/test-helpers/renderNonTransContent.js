import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
