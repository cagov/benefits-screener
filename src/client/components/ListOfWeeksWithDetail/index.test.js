import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<ListOfWeeksWithDetail />", () => {
  it("renders the component", async () => {
    const weekOfAnswers = {
      tooSick: false,
      couldNotAcceptWork: false,
      didYouLook: false,
      refuseWork: false,
      schoolOrTraining: false,
      workOrEarn: false,
    };

    const wrapper = renderNonTransContent(Component, "ListOfWeeksWithDetail", {
      userData: {
        formData: [weekOfAnswers, weekOfAnswers, weekOfAnswers],
        weeksToCertify: [1, 2, 3],
        programPlan: ["UI full time"],
      },
      showContent: [true, true, true, true],
      toggleContent: jest.fn(),
    });

    expect(wrapper).toMatchSnapshot();
  });
});
