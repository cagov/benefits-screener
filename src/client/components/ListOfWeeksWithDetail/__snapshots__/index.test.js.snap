// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfWeeksWithDetail /> renders the component 1`] = `
Array [
  <WeekWithDetail
    index={0}
    key="0"
    weekData={
      Object {
        "couldNotAcceptWork": false,
        "didYouLook": false,
        "refuseWork": false,
        "schoolOrTraining": false,
        "tooSick": false,
        "workOrEarn": false,
      }
    }
    weekIndex={1}
    weekProgramPlan="UI full time"
  />,
  <WeekWithDetail
    index={1}
    key="1"
    weekData={
      Object {
        "couldNotAcceptWork": false,
        "didYouLook": false,
        "refuseWork": false,
        "schoolOrTraining": false,
        "tooSick": false,
        "workOrEarn": false,
      }
    }
    weekIndex={2}
    weekProgramPlan="UI full time"
  />,
  <WeekWithDetail
    index={2}
    key="2"
    weekData={
      Object {
        "couldNotAcceptWork": false,
        "didYouLook": false,
        "refuseWork": false,
        "schoolOrTraining": false,
        "tooSick": false,
        "workOrEarn": false,
      }
    }
    weekIndex={3}
    weekProgramPlan="UI full time"
  />,
]
`;
